### PR TITLE
ASoC: SOF: Intel: hda-dai: skip tlv for dspless mode

### PR DESCRIPTION
**Pull request by bardliao**
sof_ipc4_dma_config_tlv{} is for Audio DSP firmware only. Don't set it in dspless mode.

Fixes: 17386cb1b48b ("ASoC: SOF: Intel: hda-dai: set dma_stream_channel_map device")

### CODE REVIEW

**Comment by bardliao** [comment]
@plbossart This commit can only be cherry pick on top of the https://github.com/thesofproject/linux/commit/17386cb1b48b0d85f69b21ae13d5408d67180e30 ("ASoC: SOF: Intel: hda-dai: set dma_stream_channel_map device") commit. But, it can not be applied with "git am". Should I use another Fixes tag that can apply with "git am"?

**Review by plbossart — DISMISSED** [review]

**Comment by plbossart** [comment]
> @plbossart This commit can only be cherry pick on top of the [17386cb](https://github.com/thesofproject/linux/commit/17386cb1b48b0d85f69b21ae13d5408d67180e30) ("ASoC: SOF: Intel: hda-dai: set dma_stream_channel_map device") commit. But, it can not be applied with "git am". Should I use another Fixes tag that can apply with "git am"?

it's fine as is, the first kernel for LNL is 6.10, so there's no problem.

File: sound/soc/sof/intel/hda-dai.c [hunk 1]
```diff
@@ -525,6 +525,9 @@ int sdw_hda_dai_hw_params(struct snd_pcm_substream *substream,
 		return ret;
 	}
 
+	if (sdev->dspless_mode_selected)
+		goto skip_tlv;
```
Comment (ujfalusi) [inline sound/soc/sof/intel/hda-dai.c]: or just `return 0;` without a new label?

**Review by ujfalusi — CHANGES_REQUESTED** [review]

File: sound/soc/sof/intel/hda-dai.c (hunk 1 above)
Comment (bardliao) [reply sound/soc/sof/intel/hda-dai.c]: I was trying to follow [skip_tlv:](https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/sof/intel/hda-dai.c#L426) in `non_hda_dai_hw_params_data`. But I am fine to just return 0. Will follow the majority.

Comment (plbossart) [reply sound/soc/sof/intel/hda-dai.c]: The initial skip_tlv isn't really useful, I am not sure why I used a goto. I would use return 0 everywhere. gotos make sense when error control gets out of hand and when the indentation level is starting to get in the way of readability. For a single level branch it should be avoided.

Comment (bardliao) [reply sound/soc/sof/intel/hda-dai.c]: Updated

**Review by plbossart — APPROVED** [review]

**Review by ujfalusi — APPROVED** [review]

**Pull request merged**